### PR TITLE
Refactor `LigerFusedLinearPreferenceBase`

### PR DESCRIPTION
**Pull request by pramodith**
## Summary
This PR refactors the `LigerFusedLinearPreferenceBase` class to contain an abstractmethod corresponding to the calculation of the loss that needs to be implemented by all sub-classes.

It also adds a new function to the class called `_compute_loss` which is mostly the same as the `_compute_orpo_loss` function introduced in #362 but makes it generic to calculate the NLL/Cross Entropy Loss plus accepts a custom loss function that implements a new alignment loss function.

Most RLHF/RLAIF/Alignment algorithms state their final loss as `NLL + Beta * (Alignment_Loss) `so adding the NLL logic inside the base class reduces repeated code.

The _compute_loss function accepts

## Testing Done

On A100-80G-SXM

- Hardware Type: <BLANK>
- [X] run `make test` to ensure correctness
- [X] run `make checkstyle` to ensure code style
- [X] run `make test-convergence` to ensure convergence

### CODE REVIEW

File: src/liger_kernel/chunked_loss/orpo_loss.py [hunk 1]
```diff
@@ -98,15 +41,18 @@ def forward(
...
-        orpo_loss_fn = partial(
-            _compute_orpo_loss,
-            full_target=target,
+
+        return LigerFusedLinearPreferenceBase.forward(
+            ctx,
+            _input,
+            weight,
+            target,
+            bias,
```
Comment (shivam15s) [inline src/liger_kernel/chunked_loss/orpo_loss.py:50]: I wrongly assumed that forward just takes positional arguments. 
Could you also make input, target, bias keyword args?

**Comment by shivam15s** [comment]
This was quick. Great refactor!

**Review by shivam15s — DISMISSED** [review]
lgtm

File: src/liger_kernel/chunked_loss/orpo_loss.py (hunk 1 above)
Comment (pramodith) [reply src/liger_kernel/chunked_loss/orpo_loss.py]: Done!

**Review by shivam15s — APPROVED** [review]
lgtm
